main.go

```diff
@@ -43,57 +45,66 @@ func main() {
 		Statmode: statmode,
 	}
 
+	var timeRemaining time.Duration
 	if *fHeight > 0 {
 		estTime, err := esti.CalcDate(*fHeight)
 		if err != nil {
-			fmt.Println(err.Error())
-			return
-		}
-		if time.Duration(estTime.UnixNano()-time.Now().UnixNano()) < *fVotingPeriod+time.Hour {
-			fmt.Println("****** WARNING ******")
-			fmt.Println("The Date above is before the voting period ends")
-			fmt.Println(time.Duration(estTime.UnixNano() - time.Now().UnixNano()).String())
-			fmt.Println("****** WARNING ******")
+			return 1, err
 		}
 
+		timeRemaining = time.Until(estTime)
+
 		fmt.Printf("Estimated time for height %d:\n", *fHeight)
 		tzs := strings.Split(*fTimezones, ",")
-		ml := 0
+		width := 0
 		for _, tz := range tzs {
-			if len(tz) >= ml {
-				ml = len(tz)
+			if len(tz) > width {
+				width = len(tz)
 			}
 		}
 		for _, tz := range tzs {
 			loc, err := time.LoadLocation(tz)
 			if err != nil {
-				fmt.Println("unable to find timezone", tz)
+				fmt.Printf("unable to find timezone %q\n", tz)
 			} else {
-				fmt.Printf("%"+fmt.Sprintf("%d", ml)+"s: %s\n", tz, estTime.In(loc).Format(time.UnixDate))
+				fmt.Printf("%*s: %s\n", width, tz, estTime.In(loc).Format(time.UnixDate))
 			}
 		}
 	} else if *fDate != "" {
 		date, err := dateparse.ParseLocal(*fDate)
 		if err != nil {
-			fmt.Println(err.Error())
-			return
+			return 1, err
```

Comment:
I wonder how much diagnostic info this loses.

Reply:
None, because the error is still printed to stderr.